tracer/src/Datadog.Trace/FeatureFlags/FeatureFlagsModule.cs

```diff
@@ -0,0 +1,101 @@
+        private ExposureApi? _exposureApi = null;
+
+        private FeatureFlagsEvaluator? _evaluator = null;
+
+        internal FeatureFlagsModule(TracerSettings settings, IRcmSubscriptionManager? rcmSubscriptionManager = null)
+        {
+            Log.Debug("FeatureFlagsModule ENABLED");
+            _rcmSubscriptionManager = rcmSubscriptionManager ?? RcmSubscriptionManager.Instance;
+            _exposureApi = ExposureApi.Create(settings);
+            _ffeProduct = new FfeProduct(UpdateRemoteConfig);
+            if (Interlocked.Exchange(ref _rcmSubscription, new Subscription(_ffeProduct.UpdateFromRcm, RcmProducts.FfeFlags)) == null)
+            {
+                _rcmSubscriptionManager.SubscribeToChanges(_rcmSubscription!);
+                _rcmSubscriptionManager.SetCapability(RcmCapabilitiesIndices.FfeFlagConfigurationRules, true);
+            }
+        }
+
+        public static FeatureFlagsModule? Create(TracerSettings settings, IRcmSubscriptionManager? rcmSubscriptionManager = null)
+        {
+            if (settings.IsFlaggingProviderEnabled)
+            {
+                return new FeatureFlagsModule(settings, rcmSubscriptionManager ?? RcmSubscriptionManager.Instance);
+            }
+
+            return null;
+        }
+
+        public void Dispose()
+        {
+            _exposureApi?.Dispose();
+        }
+
+        internal void RegisterOnNewConfigEventHandler(Action? onNewConfig)
+        {
+            _onNewConfigEventHander = onNewConfig;
+        }
+
+        internal Evaluation Evaluate(string flagKey, ValueType resultType, object? defaultValue, IEvaluationContext? context)
+        {
+            if (_evaluator is null)
+            {
+                Log.Debug("FeatureFlagsModule::Evaluate -> Evaluator is null (no config received)");
+                return new Evaluation(flagKey, null, EvaluationReason.Error, null, "No config loaded");
+            }
+
+            Log.Debug("FeatureFlagsModule::Evaluate -> Returning Evaluation");
+            return _evaluator.Evaluate(flagKey, resultType, defaultValue, context);
+        }
+
+        private void UpdateRemoteConfig(List<KeyValuePair<string, ServerConfiguration>> list)
+        {
+            Log.Debug<int>("FeatureFlagsModule::UpdateRemoteConfig -> New config received. {Count}", list.Count);
+            try
+            {
+                // Feed configs to the rules evaluator
+                if (list.Count > 0)
+                {
+                    Interlocked.Exchange(ref _evaluator, new FeatureFlagsEvaluator(ReportExposure, list[0].Value));
+                    _onNewConfigEventHander?.Invoke();
+                }
+            }
+            catch (Exception ex)
+            {
+                Log.Warning(ex, "FeatureFlagsModule::UpdateRemoteConfig -> Error processing new config");
+            }
+        }
+
+        private void ReportExposure(in ExposureEvent exposure)
+        {
+            _exposureApi?.SendExposure(exposure);
+        }
```

Comment:
nit: don't use `in` unless it's a `readonly struct`. Also `_exposureApi` should never be null here


```suggestion
        private void ReportExposure(ExposureEvent exposure)
        {
            _exposureApi.SendExposure(exposure);
        }
```

Reply:
Made ExpusureEvent a readonly record struct